azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -245,6 +244,48 @@ def _get_telemetry_topic(self):
         topic += "/messages/events/"
         return topic
 
+    def _encode_properties(self, message_to_send, topic):
+        system_properties = dict()
+        if message_to_send.message_id:
+            system_properties["$.mid"] = message_to_send.message_id
+
+        if message_to_send.correlation_id:
+            system_properties["$.cid"] = message_to_send.correlation_id
+
+        if message_to_send.user_id:
+            system_properties["$.uid"] = message_to_send.user_id
+
+        if message_to_send.to:
+            system_properties["$.to"] = message_to_send.to
+
+        if message_to_send.content_type:
+            system_properties["$.ct"] = message_to_send.content_type
+
+        if message_to_send.content_encoding:
+            system_properties["$.ce"] = message_to_send.content_encoding
+
+        if message_to_send.expiry_time_utc:
+            system_properties["$.exp"] = (
+                message_to_send.expiry_time_utc.isoformat()
+                if isinstance(message_to_send.expiry_time_utc, date)
+                else message_to_send.expiry_time_utc
+            )
+
+        # Question TODO : haven't encoded these properties here, i see it encoded in JAVA
+        # system_properties["$.cdid"]
+        # system_properties["$.cmid"]
+        if system_properties and len(system_properties) > 0:
```

Comment:
>system_properties [](start = 11, length = 17)

you initialize this yourself on line 248. it's always going to be truthy, right?